### PR TITLE
Developer mock generation

### PR DESCRIPTION
**Pull request by yarg0007**
* Introducing code to populate Map<String, Object> hierarchy according to paths and mock values provided
* Updating tool tip for mock input UI elements to explain that [*] wildcard notation will convert to [0] and default to array length 1

### CODE REVIEW

File: TMBuilder/src/main/java/com/ebay/tool/thinmodelgen/gui/menu/export/DeveloperMockExport.java [hunk 1]
```diff
@@ -0,0 +1,619 @@
...
+     * Export the validation mock values to JSON for each validation defined. Each custom validation will also include
+     * the core validation set, because, those are true in every response and therefore MUST be part of every mock.
+     * @param exportPath Folder to export mocks to.
+     * @param validationSetModels Validation set models containing mock values to write to JSON.
+     * @throws IOException Pass through.
+     * @throws ClassNotFoundException Pass through.
+     */
+    public void export(File exportPath, List<ValidationSetModel> validationSetModels) throws IOException, ClassNotFoundException {
+
+        ValidationSetModel coreValidaitonSet = getCoreValidationSet(validationSetModels);
```
Comment (juliandesmarais) [inline TMBuilder/src/main/java/com/ebay/tool/thinmodelgen/gui/menu/export/DeveloperMockExport.java:39]:
```suggestion
        ValidationSetModel coreValidationSet = getCoreValidationSet(validationSetModels);
```

File: TMBuilder/src/main/java/com/ebay/tool/thinmodelgen/gui/menu/export/DeveloperMockExport.java [hunk 2]
```diff
@@ -0,0 +1,619 @@
...
+                String[] splitPath = savedPath.split("\\.");
+                Map rootNode = jsonMap;
+                walkAndPopulateJsonMap(rootNode, splitPath, new DeveloperMockValueLooper(mockValueDecoder));
+            }
+        }
+    }
+
+    protected void walkAndPopulateJsonMap(Map currentNode, String[] pathSteps, DeveloperMockValueLooper mockValueLooper) {
+
+        if (currentNode == null || pathSteps == null || mockValueLooper == null) {
```
Comment (juliandesmarais) [inline TMBuilder/src/main/java/com/ebay/tool/thinmodelgen/gui/menu/export/DeveloperMockExport.java:133]:
```suggestion
        if (currentNode == null || pathSteps == null || pathSteps.isEmpty() || mockValueLooper == null) {
```

File: TMBuilder/src/main/java/com/ebay/tool/thinmodelgen/gui/menu/export/DeveloperMockExport.java [hunk 3]
```diff
@@ -0,0 +1,619 @@
...
+            } else {
+                return;
+            }
+        }
+
+        String pathStep = pathSteps[0];
+        // pathStepArrayIndex will be null (not an array), -1 for *, or a specific value for a specific array element.
+        Integer pathStepArrayIndex = null;
+        if (pathStep.contains("[")) {
+            String indexVal = pathStep.substring(pathStep.indexOf("[")+1, pathStep.indexOf("]"));
```
Comment (juliandesmarais) [inline TMBuilder/src/main/java/com/ebay/tool/thinmodelgen/gui/menu/export/DeveloperMockExport.java:159]: nitpick
```suggestion
            String indexVal = pathStep.substring(pathStep.indexOf("[") + 1, pathStep.indexOf("]"));
```

File: TMBuilder/src/main/java/com/ebay/tool/thinmodelgen/gui/menu/export/DeveloperMockExport.java [hunk 4]
```diff
@@ -0,0 +1,619 @@
...
+        NodeModel[] nodes = validationSetModel.getData();
+        for (NodeModel node : nodes) {
+            JsonBaseType jsonBaseType = convertNodeModelToJsonBaseType(node);
+            String[] savedPaths = jsonBaseType.getSavedPathsForNode();
+
+            for (String savedPath : savedPaths) {
+                String[] splitSavedPath = savedPath.split("\\.");
+                StringBuilder pathBuilder = new StringBuilder();
+                String lastArrayPath = "";
+                for (String value : splitSavedPath) {
```
Comment (juliandesmarais) [inline TMBuilder/src/main/java/com/ebay/tool/thinmodelgen/gui/menu/export/DeveloperMockExport.java:340]: Exception if split is empty?

File: TMBuilder/src/main/java/com/ebay/tool/thinmodelgen/gui/menu/export/DeveloperMockExport.java [hunk 5]
```diff
@@ -0,0 +1,619 @@
...
+//                    DeveloperMockValue<?> mockValue = (DeveloperMockValue<?>) jsonPathExecutor;
+//                    DeveloperMockType mockType = mockValue.getMockType();
+//                    mockValue.getMockValue();
+//                } else {
+//                    continue;
+//                }
+//            }
+//        }
+//    }
+
```
Comment (juliandesmarais) [inline TMBuilder/src/main/java/com/ebay/tool/thinmodelgen/gui/menu/export/DeveloperMockExport.java:541]: Did you mean to leave this in?

File: TMBuilder/src/main/java/com/ebay/tool/thinmodelgen/gui/menu/export/DeveloperMockExport.java [hunk 6]
```diff
@@ -0,0 +1,619 @@
...
+
+    protected void populateJsonModel(DeveloperMockListOfValues<?> mockValues, String jsonPath) {
+
+    }
```
Comment (juliandesmarais) [inline TMBuilder/src/main/java/com/ebay/tool/thinmodelgen/gui/menu/export/DeveloperMockExport.java:545]: Did you mean to leave this in?

**Review by juliandesmarais — APPROVED** [review]
Some comments

File: TMBuilder/src/main/java/com/ebay/tool/thinmodelgen/gui/menu/export/DeveloperMockExport.java (hunk 1 above)
Comment (yarg0007) [reply TMBuilder/src/main/java/com/ebay/tool/thinmodelgen/gui/menu/export/DeveloperMockExport.java:39]: Corrected.

File: TMBuilder/src/main/java/com/ebay/tool/thinmodelgen/gui/menu/export/DeveloperMockExport.java (hunk 2 above)
Comment (yarg0007) [reply TMBuilder/src/main/java/com/ebay/tool/thinmodelgen/gui/menu/export/DeveloperMockExport.java:133]: Had to use pathSteps.length == 0, but same effect. Thanks.

File: TMBuilder/src/main/java/com/ebay/tool/thinmodelgen/gui/menu/export/DeveloperMockExport.java (hunk 3 above)
Comment (yarg0007) [reply TMBuilder/src/main/java/com/ebay/tool/thinmodelgen/gui/menu/export/DeveloperMockExport.java:159]: Done.

File: TMBuilder/src/main/java/com/ebay/tool/thinmodelgen/gui/menu/export/DeveloperMockExport.java (hunk 5 above)
Comment (yarg0007) [reply TMBuilder/src/main/java/com/ebay/tool/thinmodelgen/gui/menu/export/DeveloperMockExport.java:541]: Oops. Good catch. Thanks.

File: TMBuilder/src/main/java/com/ebay/tool/thinmodelgen/gui/menu/export/DeveloperMockExport.java (hunk 6 above)
Comment (yarg0007) [reply TMBuilder/src/main/java/com/ebay/tool/thinmodelgen/gui/menu/export/DeveloperMockExport.java:545]: Nope. Removed.

**Pull request merged**